server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -197,6 +206,12 @@ public RemoteClusterStateService(
         clusterSettings.addSettingsUpdateConsumer(SLOW_WRITE_LOGGING_THRESHOLD, this::setSlowWriteLoggingThreshold);
         this.remoteStateReadTimeout = clusterSettings.get(REMOTE_STATE_READ_TIMEOUT_SETTING);
         clusterSettings.addSettingsUpdateConsumer(REMOTE_STATE_READ_TIMEOUT_SETTING, this::setRemoteStateReadTimeout);
+        this.checksumValidationEnabled = clusterSettings.get(REMOTE_CLUSTER_STATE_CHECKSUM_VALIDATION_ENABLED_SETTING);
```

Comment:
I was wondering instead of a boolean setting we have an enum with the below modes for validations
1. monitor_only : we only log checksum mismatches but don't fail anything
2. defensive : we fail with an `IllegalStateException`

Reply:
PR to address this https://github.com/opensearch-project/OpenSearch/pull/15622